dockerfiles/base/scripts/base/cli/cli-functions.sh

```diff
@@ -188,6 +185,27 @@ grab_offline_images(){
       fi
       info "init" "Loading ${file##*/}..."
     done
+  else
+    # If we are here, then we want to run in networking mode.
+    # If we are in networking mode, we have had some issues where users have failed DNS networking.
+    # See: https://github.com/eclipse/che/issues/3266#issuecomment-265464165
+    local HTTP_STATUS_CODE=$(curl -I -k dockerhub.com -s -o /dev/null --write-out "%{http_code}")
```

Comment:
This check doesn't work behind the proxy. Need to export HTTP_PROXY variables or smthn like that.